### DISCUSSION TITLE
HOW DO I USE AS A PORTABLE THAT CAN BE MOVED FROM ENVIRONMENT TO ENVIRONMENT?

### DISCUSSION THREAD

**Discussion by ATOMPEAR** [JanDeDobbeleer/oh-my-posh#5918] [General]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

# HERES THE PROBLEM AND ITS NOT A BUG :) I NEED HELP
SO I HAVE AN APPLICATION IM DEVELOPING. IT IS DISTRIBUTED WITH WINDOWS TERMINAL FROM GITHUB. THERE IS A ZIPPED VERSION THAT IS PORTABLE. I HAVE ALSO USE CMDER AND CMDER IN EXPERIMENTS SO THAT CAN BE DONE IF IT WOULD BE A BETTER FIT FOR FINDING THE SOLUTIONS. I WANT TO CUSTOMIZE MY PORTABLE TERMINAL THAT IS DISTRIBUTED WITH MY APPLICATION WITH OH-MY-POSH AND I WANT THAT APPLICATION TO BE ABLE TO MOVE FROM ENVIROMENT TO ENVIROMENT WITHOUT HAVING TO DO TO MUCH THAT WILL FEEL CUMBERSOME TO HAVE TO SET UP EVER TIME THE ENVIROMENT IS MOVED. IVE CONCIDER SPACESHIP TO THAT MIGHT BE BETTER FOR THIS PURPOSE IM NOT SURE, BUT I PREFURE OH-MY-POSH. 

### NEXT HOW THIS WORK?
IVE DOWNLOADED POSH-WINDOWS-386.EXE (<--32BIT I THINK?) AND POSH-WINDOWS-AMD64.EXE AS WELL AS THEMES.ZIP AND EVEN OH-MY-POSH.JSON THIS IS ALL I NEED TO ACHIVE WHAT IM TRYING TO DO CORRECT? I THINK I HAVE TO ADD THE EXECUTABLES TO THE PATH SOMEHOW RIGHT? OR SOMETHING ELSE? WILL I HAVE TO DO THAT EVERY TIME MY APPLICATION WITH THE OH-MY-PSH CUSTOMIZED TERMINAL MOVED TO A NEW ENVIROMENT? I WOULD PREFER NOT BUT IF THAT IS THE CASE CAN I FIX THE PATH SO IT WORK WHEN MOVED TO A NEW ENVIROMENT WITH A POWERSHELL SCRIPT. THEN I COULD MAKE THE POWSERSHELL SCRIPT AND EXE OR SOMETHING AND IT WOULDNT BE THAT COMBERSOME TO DO WHEN THE EVIROMENT IS SWITCHED.

### Theme

NA

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
NA
```

**Comment by ATOMPEAR** [comment]


**Reply by JanDeDobbeleer** [reply]
@ATOMPEAR all you need to do is add the location of the oh-my-posh executable's location (so `POSH-WINDOWS-AMD64.EXE`) to your `PATH` on the machine you're using. This needs to happen in every environment, and can also be done from a PowerShell script that's added to your portable setup so it can be dynamic. It's that and your configuration, which can also be resolved dynamically from the same PowerShell script as you need to have a portable `$PROFILE` anyways to load oh-my-posh.

**Reply by ATOMPEAR** [reply]
What do you mean by portable $PROFILE? You lost me a little there. I've been messing around. I currently am right now and haven't quite figured it out. I'm getting closer though lol

**Reply by JanDeDobbeleer** [reply]
@ATOMPEAR https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_profiles?view=powershell-7.4
